x-pack/plugins/security_solution/public/exceptions/api/list_api.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { fetchExceptionLists, updateExceptionList } from '@kbn/securitysolution-list-api';
+
+import type { HttpSetup } from '@kbn/core-http-browser';
+import { getFilters } from '@kbn/securitysolution-list-utils';
+import type { List, ListArray } from '@kbn/securitysolution-io-ts-list-types';
+import { ALL_ENDPOINT_ARTIFACT_LIST_IDS } from '../../../common/endpoint/service/artifacts/constants';
+import type {
+  FetchListById,
+  LinkListToRules,
+  UnlinkListFromRules,
+  UpdateExceptionList,
+} from './types';
+import { fetchRules, patchRule } from '../../detection_engine/rule_management/api/api';
+import type { Rule } from '../../detection_engine/rule_management/logic';
+
+export const getListById = async ({ id, http }: FetchListById) => {
+  try {
+    const abortCtrl = new AbortController();
+    const filters = getFilters({
+      filters: { list_id: id },
+      namespaceTypes: ['single', 'agnostic'],
+      hideLists: ALL_ENDPOINT_ARTIFACT_LIST_IDS,
+    });
+    const namespaceTypes = ['single', 'agnostic'].join();
+
+    const { data } = await fetchExceptionLists({
+      filters,
+      http: http as HttpSetup,
+      signal: abortCtrl.signal,
+      namespaceTypes,
+      pagination: {},
+    });
+    abortCtrl.abort();
+
+    if (data && data.length) return data[0];
+    return null;
+  } catch (error) {
+    throw new Error(error);
+  }
+};
+export const getListRules = async (listId: string) => {
+  try {
+    const abortCtrl = new AbortController();
+    const { data: rules } = await fetchRules({
+      signal: abortCtrl.signal,
+    });
+    abortCtrl.abort();
+    return rules.reduce((acc: Rule[], rule, index) => {
+      const listExceptions = rule.exceptions_list?.find(
+        (exceptionList) => exceptionList.list_id === listId
+      );
+      if (listExceptions) acc.push(rule);
+      return acc;
```

Comment:
Discussed yesterday but posting here for reference - (at a later date) we should investigate if we can add backwards saved object references (i.e. `exception_list.references = [rule_id1, rule_id2, ....]`) which would remove the need for building these relationships in memory.

Reply:
Yes, Thanks @dhurley14 ^_^